code/modules/mod/mod_modules.dm

```diff
@@ -0,0 +1,616 @@
+	storage.max_w_class = max_w_class
+	storage.max_combined_w_class = max_combined_w_class
+	storage.max_items = max_items
+
+/obj/item/mod/module/storage/on_install()
+	var/datum/component/storage/modstorage = mod.AddComponent(/datum/component/storage, storage)
+	modstorage.max_w_class = max_w_class
+	modstorage.max_combined_w_class = max_combined_w_class
+	modstorage.max_items = max_items
+
+/obj/item/mod/module/storage/on_uninstall()
+	var/datum/component/storage/modstorage = mod.GetComponent(/datum/component/storage)
+	modstorage.RemoveComponent()
+
+/obj/item/mod/module/visor
+	name = "MOD visor module"
+	desc = "A module installed to the helmet, allowing access to different views."
+	module_type = MODULE_TOGGLE
+	complexity = 3
+	active_power_cost = 10
+	incompatible_modules = list(/obj/item/mod/module/visor)
+	cooldown_time = 0.5 SECONDS
+	var/helmet_tint = 0
+	var/helmet_flash_protect = FLASH_PROTECTION_NONE
+	var/hud_type = null
+	var/list/visor_traits = list()
+
+/obj/item/mod/module/visor/on_activation()
+	. = ..()
+	if(!.)
+		return
+	mod.helmet.tint = helmet_tint
+	mod.helmet.flash_protect = helmet_flash_protect
+	if(hud_type)
+		var/datum/atom_hud/hud = GLOB.huds[hud_type]
+		hud.add_hud_to(mod.wearer)
+	for(var/trait in visor_traits)
+		ADD_TRAIT(mod.wearer, trait, MOD_TRAIT)
+	mod.wearer.update_sight()
+	mod.wearer.update_tint()
+
+/obj/item/mod/module/visor/on_deactivation()
+	. = ..()
+	if(!.)
+		return
+	mod.helmet.tint = initial(mod.helmet.tint)
+	mod.helmet.flash_protect = initial(mod.helmet.flash_protect)
+	if(hud_type)
+		var/datum/atom_hud/hud = GLOB.huds[hud_type]
+		hud.remove_hud_from(mod.wearer)
+	for(var/trait in visor_traits)
+		REMOVE_TRAIT(mod.wearer, trait, MOD_TRAIT)
+	mod.wearer.update_sight()
+
+/obj/item/mod/module/visor/medhud
+	name = "MOD medical visor module"
+	hud_type = DATA_HUD_MEDICAL_ADVANCED
+	visor_traits = list(TRAIT_MEDICAL_HUD)
+
+/obj/item/mod/module/visor/diaghud
+	name = "MOD medical visor module"
+	hud_type = DATA_HUD_DIAGNOSTIC_ADVANCED
+	visor_traits = list(TRAIT_DIAGNOSTIC_HUD)
+
+/obj/item/mod/module/visor/sechud
+	name = "MOD medical visor module"
+	hud_type = DATA_HUD_SECURITY_ADVANCED
+	visor_traits = list(TRAIT_SECURITY_HUD)
+
+/obj/item/mod/module/visor/welding
+	name = "MOD welding visor module"
+	helmet_tint = 2
+	helmet_flash_protect = FLASH_PROTECTION_WELDER
+
+/obj/item/mod/module/visor/sunglasses
+	name = "MOD protective visor module"
+	helmet_tint = 1
+	helmet_flash_protect = FLASH_PROTECTION_FLASH
+
+/obj/item/mod/module/visor/meson
+	name = "MOD meson visor module"
+	visor_traits = list(TRAIT_MESON_VISION)
+
+/obj/item/mod/module/health_analyzer
+	name = "MOD health analyzer module"
+	desc = "A module with a microchip health analyzer to instantly scan the wearer's vitals."
+	module_type = MODULE_USABLE
+	complexity = 1
+	use_power_cost = 30
+	incompatible_modules = list(/obj/item/mod/module/health_analyzer)
+	cooldown_time = 0.5 SECONDS
+	var/module_advanced = FALSE
```

Comment:
are you... gonna add a subtype of this that has a module_advanced of TRUE?

Reply:
who knows